### PR TITLE
fix: remove btc account from permission connect lists

### PR DESCRIPTION
**Pull request by montelaidev**
## **Description**

This PR filters BTC accounts from the permission connect list.

## **Related issues**

Fixes: 

## **Manual testing steps**

1. Create a BTC account
2. Connect to a dapp
3. Open popup
4. Click network icon
5. Click on `Connect more accounts`
6. See that there aren't any btc accounts

### **Before**

Screenshot 2024-07-19 at 21 09 57

### **After**

Screenshot 2024-07-19 at 21 04 04

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/connect-accounts-modal/__snapshots__/connect-accounts-modal.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Connect More Accounts Modal should render correctly 1`] = `<div />`;
```
Comment (ccharly) [inline ui/components/multichain/connect-accounts-modal/__snapshots__/connect-accounts-modal.test.tsx.snap:3]: This feels kinda empty, is this normal?

File: ui/components/multichain/connect-accounts-modal/connect-accounts-modal.tsx [hunk 2]
```diff
@@ -13,7 +14,9 @@ export const ConnectAccountsModal = ({
   activeTabOrigin: string;
 }) => {
   const accounts = useSelector((state) =>
-    getUnconnectedAccounts(state, activeTabOrigin),
+    getUnconnectedAccounts(state, activeTabOrigin).filter(
```
Comment (ccharly) [inline ui/components/multichain/connect-accounts-modal/connect-accounts-modal.tsx:17]: I feel like we should explain why we are filtering them out (you can update the wording of my suggestion, it was just to give an example):

```suggestion
    // We only consider EVM accounts.
    // Connections with non-EVM accounts (Bitcoin only for now) are used implicitly and handled by the Bitcoin Snap itself.
    getUnconnectedAccounts(state, activeTabOrigin).filter(
```

File: ui/pages/permissions-connect/permissions-connect.container.js [hunk 3]
```diff
@@ -84,7 +85,9 @@ const mapStateToProps = (state, ownProps) => {
 
   const requestState = getRequestState(state, permissionsRequestId) || {};
 
-  const accountsWithLabels = getAccountsWithLabels(state);
+  const accountsWithLabels = getAccountsWithLabels(state).filter((account) =>
```
Comment (ccharly) [inline ui/pages/permissions-connect/permissions-connect.container.js:88]: We should use the same comment here (you can update the wording of my suggestion, it was just to give an example):

```suggestion
    // We only consider EVM accounts.
    // Connections with non-EVM accounts (Bitcoin only for now) are used implicitly and handled by the Bitcoin Snap itself.
  const accountsWithLabels = getAccountsWithLabels(state).filter((account) =>
```

File: ui/pages/permissions-connect/permissions-connect.test.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+const render = (
+  props = defaultProps,
+  type: ApprovalType = ApprovalType.WalletRequestPermissions,
+) => {
+  let pendingPermission;
+  if (type === ApprovalType.WalletRequestPermissions) {
+    pendingPermission = {
+      id: mockPermissionRequestId,
+      origin: 'https://.metamask.io',
```
Comment (ccharly) [inline ui/pages/permissions-connect/permissions-connect.test.tsx:57]: IDK if the `.` was intended?

```suggestion
      origin: 'https://metamask.io',
```

File: ui/pages/permissions-connect/permissions-connect.test.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      type: ApprovalType.WalletRequestPermissions,
+      time: 1721376328642,
+      requestData: {
+        metadata: {
+          id: mockPermissionRequestId,
+          origin: 'https://.metamask.io',
```
Comment (ccharly) [inline ui/pages/permissions-connect/permissions-connect.test.tsx:63]:
```suggestion
          origin: 'https://metamask.io',
```

File: ui/pages/permissions-connect/permissions-connect.test.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  return renderWithProvider(
+    <PermissionApprovalContainer {...props} />,
+    store,
+    `${CONNECT_ROUTE}/${mockPermissionRequestId}`,
+  );
+};
+
+describe('PermissionApprovalContainer', () => {
+  describe('ConnectPath', () => {
+    it('should render  correctly', () => {
```
Comment (ccharly) [inline ui/pages/permissions-connect/permissions-connect.test.tsx:122]: Nit:

```suggestion
    it('should render correctly', () => {
```

**Review by ccharly — DISMISSED** [review]
LGTM (left 1 nit, which is really minor)

Also tested on my side:

Screenshot 2024-07-25 at 14 53 03
Screenshot 2024-07-25 at 14 53 09
Screenshot 2024-07-25 at 14 53 36
Screenshot 2024-07-25 at 14 53 55
Screenshot 2024-07-25 at 14 54 06

File: test/e2e/flask/btc/btc-dapp-connection.spec.ts [hunk 7]
```diff
@@ -32,15 +31,9 @@ describe('BTC Account - Dapp Connection', function (this: Suite) {
         await driver.waitUntilXWindowHandles(3);
         await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-        const account2 = await driver.waitForSelector(
+        await driver.assertElementNotPresent(
           '[data-testid="choose-account-list-1"]',
```
Comment (danroc) [inline test/e2e/flask/btc/btc-dapp-connection.spec.ts:35]: This is the ID of the Bitcoin account.

File: ui/pages/permissions-connect/permissions-connect.test.tsx (hunk 6 above)
Comment (danroc) [inline ui/pages/permissions-connect/permissions-connect.test.tsx:122]:
```suggestion
    it('renders correctly', () => {
```

File: ui/pages/permissions-connect/permissions-connect.test.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+      const { container, getByText } = render();
+      expect(getByText(messages.next.message)).toBeInTheDocument();
+      expect(getByText(messages.cancel.message)).toBeInTheDocument();
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should render list without BTC accounts', async () => {
```
Comment (danroc) [inline ui/pages/permissions-connect/permissions-connect.test.tsx:129]:
```suggestion
    it('renders the list without BTC accounts', async () => {
```

**Review by danroc — DISMISSED** [review]

**Review by ccharly — DISMISSED** [review]

**Review by ccharly — APPROVED** [review]

**Review by danroc — APPROVED** [review]

**Pull request merged**